spark/README.md

```diff
@@ -1,158 +1,143 @@
-- Apache Spark package newer than v3.4.0 is downloaded (see the [Download Apache Spark](https://spark.apache.org/downloads.html) to download Apache Spark package )
-- Spark is configured with enough spark driver memory before starting the Spark Connect Server. Add the following configuration to the `./conf/spark-defaults.conf` in apache spark package.
+## Prerequisites
 
-```shell
-spark.driver.memory 6g
-```
-
-- Spark Connect Server is running locally (refer to the [Quickstart: Spark Connect](https://spark.apache.org/docs/latest/api/python/getting_started/quickstart_connect.html) to launch spark server with spark connect)
-- Install [Spark dependencies](https://spark.apache.org/docs/latest/api/python/getting_started/install.html#dependencies) in a dedicated python virtual environment.
-
-1. Initialise a Spice app
-
-   ```shell
-   spice init spark_demo
-   cd spark_demo
-   ```
-
-2. Start the Spice runtime
+This recipe requires [Docker](https://www.docker.com/) and [Docker Compose](https://docs.docker.com/compose/) to be installed.
 
-   ```shell
-   >>> spice run
-   Spice.ai runtime starting...
-   2024-05-20T23:54:42.323695Z  INFO spiced: Metrics listening on 127.0.0.1:9090
-   2024-05-20T23:54:42.325278Z  INFO runtime::opentelemetry: Spice Runtime OpenTelemetry listening on 127.0.0.1:50052
-   2024-05-20T23:54:42.327243Z  INFO runtime::http: Spice Runtime HTTP listening on 127.0.0.1:8090
-   2024-05-20T23:54:42.327255Z  INFO runtime::flight: Spice Runtime Flight listening on 127.0.0.1:50051
-   ```
+Also ensure that you have the `spice` CLI installed. You can find instructions on how to install it [here](https://docs.spiceai.org/getting-started).
 
-3. Create the Sample Dataset in Spark
+## How to run
 
-   This Quickstarts use NYC taxi trip parquet data from [TLC Trip Record Data](https://www.nyc.gov/site/tlc/about/tlc-trip-record-data.page) to create a sample table in Spark.
+Clone this cookbook repo locally and navigate to the `spark` directory:
 
-   Download the NYC taxi trip parquet file using the following command
-
-   ```shell
-   wget https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2024-01.parquet
-   ```
-
-   Change the `parquet_file_path` in python script to the absolute file path where `yellow_tripdata_2024-01.parquet` is located. Run the following python script in the python virtual environment that already have [Spark dependencies](https://spark.apache.org/docs/latest/api/python/getting_started/install.html#dependencies) downloaded.
+```bash
+git clone https://github.com/spiceai/cookbook.git
+cd cookbook/spark
+```
 
-   ```python
-   from pyspark.sql import SparkSession
-   from pyspark.sql import Row
+1. Start the Docker Compose stack, which includes a Spark instance and init notebook to load the NYC taxi trip parquet data:
 
-   SparkSession.builder.master("local[*]").getOrCreate().stop()
-   spark = SparkSession.builder.remote("sc://localhost:15002").getOrCreate()
+  ```shell
+  make
```

Comment:
We might want to duplicate alternative command (docker compose up -d) in case make is not installed

Reply:
updated